substdio.h

```diff
@@ -1,22 +1,46 @@
 #ifndef SUBSTDIO_H
 #define SUBSTDIO_H
 
+#include <sys/types.h>
+#include <unistd.h>
+
 typedef struct substdio {
   char *x;
   int p;
   int n;
   int fd;
-  int (*op)();
+  union {
+    ssize_t (*op)(int,void*,size_t);
+    ssize_t (*opw)(int,const void*,size_t);
+  };
```

Comment:
It appears we could avoid (or defer) the need for `SUBSTDIO_FDBUFW`, and the considerable associated churn, by not introducing these prototypes.

What would we lose (or defer) if we instead replaced `int (*op)()` with `ssize_t (*op)()`?

Reply:
I was under the impression that this fixes a warning that I have seen somewhere, but it looks like it is actually not the case. Will drop this.

Reply:
Woohoo, thank you! This makes a huge difference in the invasiveness of the patch.